### PR TITLE
fix: flaky test `Confirmation Redesign ERC721 Approve Component Submit an Approve transaction @no-mmi Sends a type 2 transaction (EIP1559)`

### PR DESCRIPTION
**Pull request by seaona**
## **Description**

Removing anti-patterns from erc721 tests.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27928?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/tokens/nft/erc721-interaction.spec.js [hunk 1]
```diff
@@ -51,19 +51,17 @@ describe('ERC721 NFTs testdapp interaction', function () {
         await driver.clickElement(
           '[data-testid="account-overview__activity-tab"]',
         );
-        const transactionItem = await driver.waitForSelector({
+        await driver.waitForSelector({
           css: '[data-testid="activity-list-item-action"]',
           text: 'Deposit',
         });
-        assert.equal(await transactionItem.isDisplayed(), true);
```
Comment (seaona) [inline test/e2e/tests/tokens/nft/erc721-interaction.spec.js:58]: anti-pattern: this causes race conditions when the transaction element is found and updated before we do the assertion, making the element become stale. See example error of this:

Screenshot from 2024-10-17 19-57-08

File: test/e2e/tests/tokens/nft/erc721-interaction.spec.js [hunk 2]
```diff
@@ -51,19 +51,17 @@ describe('ERC721 NFTs testdapp interaction', function () {
...
 
         // verify the mint transaction has finished
         await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
-        const nftsMintStatus = await driver.findElement({
+        await driver.waitForSelector({
           css: '#nftsStatus',
           text: 'Mint completed',
         });
-        assert.equal(await nftsMintStatus.isDisplayed(), true);
```
Comment (seaona) [inline test/e2e/tests/tokens/nft/erc721-interaction.spec.js:66]: anti-pattern

File: test/e2e/tests/tokens/nft/erc721-interaction.spec.js [hunk 3]
```diff
@@ -116,11 +114,10 @@ describe('ERC721 NFTs testdapp interaction', function () {
         await driver.clickElement(
           '[data-testid="account-overview__activity-tab"]',
         );
-        const transactionItem = await driver.waitForSelector({
+        await driver.waitForSelector({
           css: '[data-testid="activity-list-item-action"]',
           text: 'Deposit',
         });
-        assert.equal(await transactionItem.isDisplayed(), true);
```
Comment (seaona) [inline test/e2e/tests/tokens/nft/erc721-interaction.spec.js:123]: anti-pattern

File: test/e2e/tests/tokens/nft/erc721-interaction.spec.js [hunk 4]
```diff
@@ -474,11 +467,10 @@ describe('ERC721 NFTs testdapp interaction', function () {
         );
 
         // Verify transaction
-        const completedTx = await driver.waitForSelector({
+        await driver.waitForSelector({
           css: '[data-testid="activity-list-item-action"]',
           text: 'Approve TDN with no spend limit',
         });
-        assert.equal(await completedTx.isDisplayed(), true);
```
Comment (seaona) [inline test/e2e/tests/tokens/nft/erc721-interaction.spec.js:481]: anti-pattern

File: test/e2e/tests/tokens/nft/erc721-interaction.spec.js [hunk 5]
```diff
@@ -544,11 +536,10 @@ describe('ERC721 NFTs testdapp interaction', function () {
...
```
Comment (seaona) [inline test/e2e/tests/tokens/nft/erc721-interaction.spec.js:551]: anti-pattern

**Review by DDDDDanica — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM !

**Review by hjetpoluru — APPROVED** [review]
LGTM

**Pull request merged**